r/aws_eks_addon: Acceptance test failure - TestAccAWSEksAddon_AddonVersion

<!---
Please note the following potential times when an issue might be in Terraform core:

* [Configuration Language](https://www.terraform.io/docs/configuration/index.html) or resource ordering issues
* [State](https://www.terraform.io/docs/state/index.html) and [State Backend](https://www.terraform.io/docs/backends/index.html) issues
* [Provisioner](https://www.terraform.io/docs/provisioners/index.html) issues
* [Registry](https://registry.terraform.io/) issues
* Spans resources across multiple providers

If you are running into one of these scenarios, we recommend opening an issue in the [Terraform core repository](https://github.com/hashicorp/terraform/) instead.
--->

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

`TestAccAWSEksAddon_AddonVersion` fails with

```
=== RUN   TestAccAWSEksAddon_AddonVersion
=== PAUSE TestAccAWSEksAddon_AddonVersion
=== CONT  TestAccAWSEksAddon_AddonVersion
resource_aws_eks_addon_test.go:187: Step 1/3 error: Error running apply: exit status 1
Error: error creating EKS Add-On (tf-acc-test-1269590269164588671:vpc-cni): InvalidParameterException: Addon version specified is not supported
{
RespMetadata: {
StatusCode: 400,
RequestID: "e0c22503-d18d-494d-b0ce-4548a0b90c04"
},
AddonName: "vpc-cni",
ClusterName: "tf-acc-test-1269590269164588671",
Message_: "Addon version specified is not supported"
}
on terraform_plugin_test.tf line 70, in resource "aws_eks_addon" "test":
70: resource "aws_eks_addon" "test" {
--- FAIL: TestAccAWSEksAddon_AddonVersion (690.42s)
```

--- a/.changelog/20562.txt
+++ b/.changelog/20562.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/aws_eks_addon: Treat `DEGRADED` as a pending state during creation
+```

--- a/aws/internal/service/eks/waiter/waiter.go
+++ b/aws/internal/service/eks/waiter/waiter.go
@@ -19,7 +19,7 @@ const (
 
 func AddonCreated(ctx context.Context, conn *eks.EKS, clusterName, addonName string) (*eks.Addon, error) {
 	stateConf := resource.StateChangeConf{
-		Pending: []string{eks.AddonStatusCreating},
+		Pending: []string{eks.AddonStatusCreating, eks.AddonStatusDegraded},
 		Target:  []string{eks.AddonStatusActive},
 		Refresh: AddonStatus(ctx, conn, clusterName, addonName),
 		Timeout: AddonCreatedTimeout,

--- a/aws/resource_aws_eks_addon_test.go
+++ b/aws/resource_aws_eks_addon_test.go
@@ -180,8 +180,8 @@ func TestAccAWSEksAddon_AddonVersion(t *testing.T) {
 	rName := acctest.RandomWithPrefix("tf-acc-test")
 	resourceName := "aws_eks_addon.test"
 	addonName := "vpc-cni"
-	addonVersion1 := "v1.6.3-eksbuild.1"
-	addonVersion2 := "v1.7.5-eksbuild.1"
+	addonVersion1 := "v1.8.0-eksbuild.1"
+	addonVersion2 := "v1.9.0-eksbuild.1"
 	ctx := context.TODO()
 
 	resource.ParallelTest(t, resource.TestCase{

